bot-twitter/src/keeper.ts

```diff
@@ -0,0 +1,71 @@
+import { AuctionInitialInfo } from 'auctions-core/src/types';
+import getSigner from 'auctions-core/src/signer';
+import { bidOnTheAuction, enrichAuction } from 'auctions-core/src/auctions';
+import {
+    authorizeCollateral,
+    authorizeWallet,
+    getCollateralAuthorizationStatus,
+    getWalletAuthorizationStatus,
+} from 'auctions-core/src/authorizations';
+import { ETHEREUM_NETWORK, KEEPER_MIN_PROFIT_DAI } from './variables';
+
+async function participate(auction: AuctionInitialInfo, clearCache = false) {
+    const signer = await getSigner(ETHEREUM_NETWORK);
+
+    // enrich the auction with more numbers
+    const auctionTransaction = await enrichAuction(ETHEREUM_NETWORK, auction);
```

Comment:
As discussed in a call, can you please validate that calling this function once auction is finished will result in the recursion termination?

Example edge case: while `authorizeWallet` is running, the auctions was taken by someone else.
The way to test: execute `enrichAuction` on a long time finished auction to see what it returns / if it throws any errors

Reply:
How exactly can I get a long time finished auction?

Reply:
> ```js
> await enrichAuction(ETHEREUM_NETWORK, {...auction, collateralType: 'ETH-A', auctionId: 1})
> ```

No error messages or issues when executing this! Therefore I assume it can handle even finished Auctions.